### PR TITLE
feat(node-ui): new application login popup

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(node-ui): new application login popup

## Summary
This login process will be used by frontends for installed applications. (e.g. only peers, app template).
Missing parts in this is handler from the SDK that will be used by apps itself.

When user is redirected to admin dashboard (with application id entered in the frontend app) he is shown login popup -> select context -> select identity -> create jwt -> redirected to frontend (callbackURL)

Last part in video missing as I'm waiting for JWT pr to be merged.

Fixes # [Issue](https://github.com/calimero-network/core/issues/571)

## Test plan

https://github.com/user-attachments/assets/6241bf6c-734e-487f-b667-e6a776e62dc9

### CODE REVIEW

**Comment by frdomovic** [comment]
Additions:
In useEffect of the popup wrapper the setup popup function executes every time url changes. This is needed to display popup even when we changes pages.

In the video next examples are shown:
1. When user is redirected to admin dashboard and he is not logged in, he is redirected to / (setup node url page) and url params (app id + callback url) are saved in the localStorage. On every page in login process isAuthorized is checked so it can display popup with params fetch from local storage. After user is logged in he is redirected to /identity (start page when logged in) and display is shown with params from local storage. 
-> Local storage is needed so that login params are not lost in pages changes in login and redirect process.

2. When user is redirected to admin dashboard and he is logged in then the popup is displayed directly from url params. 

The finish process of login fetches admin api endpoint to create JWT token and when that data is fetched local storage params ( app id + callback url) are removed and user is redirected back to callback url with JWT in query params.

The issue with admin-dashboard not found is fixed in additional last commit. cc @alenmestrov 


https://github.com/user-attachments/assets/f5041084-f06a-436d-9866-4f1d55e02f7b

File: node-ui/src/api/dataSource/NodeDataSource.ts [hunk 1]
```diff
@@ -501,4 +514,41 @@ export class NodeDataSource implements NodeApi {
...
   }
+
+  async getContextIdentity(
+    contextId: string,
+  ): ApiResponse<ContextIdentitiesResponse> {
+    const headers: Header | null = await createAuthHeader(
+      JSON.stringify(contextId),
+    );
+    if (!headers) {
+      return { error: { code: 401, message: 'Unauthorized' } };
```
Comment (alenmestrov) [inline node-ui/src/api/dataSource/NodeDataSource.ts]: translations

File: node-ui/src/api/dataSource/NodeDataSource.ts [hunk 2]
```diff
@@ -501,4 +514,41 @@ export class NodeDataSource implements NodeApi {
...
+
+  async createAccessToken(
+    contextId: string,
+    contextIdentity: string,
+  ): ApiResponse<CreateTokenResponse> {
+    const headers: Header | null = await createAuthHeader(
+      JSON.stringify(contextId),
+    );
+    if (!headers) {
+      return { error: { code: 401, message: 'Unauthorized' } };
```
Comment (alenmestrov) [inline node-ui/src/api/dataSource/NodeDataSource.ts]: translations

**Review by alenmestrov — APPROVED** [review]

File: node-ui/src/api/dataSource/NodeDataSource.ts [hunk 3]
```diff
@@ -501,4 +514,41 @@ export class NodeDataSource implements NodeApi {
...
+  ): ApiResponse<CreateTokenResponse> {
+    const headers: Header | null = await createAuthHeader(
+      JSON.stringify(contextId),
+    );
+    if (!headers) {
+      return { error: { code: 401, message: t.unauthorizedErrorMessage } };
+    }
+
+    return await this.client.post<CreateTokenResponse>(
+      `${getAppEndpointKey()}/admin-api/contexts/${contextId}/identities`,
```
Comment (frdomovic) [inline node-ui/src/api/dataSource/NodeDataSource.ts]: fix

**Pull request merged**